src/spikeinterface/preprocessing/zero_channel_pad.py

```diff
@@ -1,10 +1,80 @@
-from typing import Union
+from typing import List, Union
 import numpy as np
 from spikeinterface.core import BaseRecording, BaseRecordingSegment
 from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
 from spikeinterface.core.core_tools import define_function_from_class
 
 
+class TracePaddedRecording(BasePreprocessor):
+    """
+    A Pre-processor class to lazily pad recordingss.
+
+    The class retrieves traces from the parent recording segment and pads them with zeros at both ends.
+
+    Parameters
+    ----------
+    parent_recording_segment : BaseRecordingSegment
+        The parent recording segment from which the traces are to be retrieved.
+    padding_left : int
+        The amount of padding to add to the left of the traces.
+    padding_right : int
+        The amount of padding to add to the right of the traces.
+    """
+
+    def __init__(self, parent_recording: BaseRecording, padding_left: int = 0, padding_right: int = 0):
+        super().__init__(recording=parent_recording)
+
+        self.padding_left = padding_left
+        self.padding_right = padding_right
+        for segment in parent_recording._recording_segments:
+            recording_segment = TracePaddedRecordingSegment(
+                segment, parent_recording.get_num_channels(), self.dtype, self.padding_left, self.padding_right
+            )
+            self.add_recording_segment(recording_segment)
+
+        self._kwargs = dict(parent_recording=parent_recording, padding_left=padding_left, padding_right=padding_right)
+
+
+class TracePaddedRecordingSegment(BasePreprocessorSegment):
+    def __init__(
+        self, parent_recording_segment: BaseRecordingSegment, num_channels, dtype, paddign_left, padding_right
+    ):
+        self.padding_left = paddign_left
+        self.padding_right = padding_right
+        self.num_channels = num_channels
+        self.dtype = dtype
+
+        super().__init__(parent_recording_segment=parent_recording_segment)
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+
+        trace_size = end_frame - start_frame
+        output_traces = np.zeros((trace_size, self.num_channels), dtype=self.dtype)
+
+        # If the stat_frame is shorter than padding left then we add the zeros to till start_frame is reached
```

Comment:
Thanks a bunch!
I improved the documentation a bit in the last commit.

Not that is your reponsablity but you know you can make suggestions by doing the following:

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/5c9e140f-a35f-419a-9749-bc4fac148bc4)

Just passing a tip in case you did not know.

Reply:
Ah so this is how you do the magic suggestions!
And here I thought it was because I wasn't an admin in the repo ^^'

Thanks for the tip!